### PR TITLE
🐛Bugfix/644/listing makes pennsieve client fail

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

The connection with Pennsieve tends to fail after several calls to large datasets fail. This is an attempt to fix the problem.

- [x] remove dependency to pennsieve client, now authenticating directly using [pennsieve recipes](https://docs.pennsieve.io/recipes) and boto3/raw client
- [x] remove dependency to pennsieve agent since we do not upload at the moment
- [x] upgrades to later best practices in datcore-adapter fastapi application
- [x] added 5 minutes caching to list files in a dataset
- [x] increased default timeout in storage/webserver clients since datasets in pennsieve tends to be large (a better fix would be using long running tasks and/or not listing everything inside a dataset as this is rather costly in performance)
- [x] now using cancel_on_disconnect decorator from servicelib instead of locally defined one
- [x] fixed loglevel parsing, added classic middlewares
- [x] added Makefile recipe to do real unit testing against pennsieve.io API

### Bonus:
- aiohttp now reports 504 Gateway Timeout error when the internal client times-out on a forwarded call to storage or another service instead of 500.

## Related issue/s
related to ITISFoundation/osparc-issues#644
related to https://github.com/ITISFoundation/dockerfiles/pull/72

### CODE REVIEW

File: services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
+
 
 @router.get(
     "/datasets",
     summary="list datasets",
     status_code=status.HTTP_200_OK,
     response_model=Page[DatasetsOut],
 )
-@cancellable_request
+@cancel_on_disconnect
```
Comment (pcrespov) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py:29]: cool! thx for this update!

File: services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py [hunk 2]
```diff
@@ -128,51 +115,25 @@ async def list_dataset_collection_files(
     return create_page(items=file_metas, total=total, params=params)
 
 
-@router.post(
-    "/datasets/{dataset_id}/files/{collection_id}",
-    summary="uploads a file into a collection",
-    status_code=status.HTTP_202_ACCEPTED,
-)
-@cancellable_request
-async def upload_file_in_collection(
```
Comment (pcrespov) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py:137]: so these handlers are not anymore used? I do not see a replacement

File: services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py [hunk 3]
```diff
@@ -22,24 +25,25 @@
 def create_app(settings: Optional[Settings] = None) -> FastAPI:
     if settings is None:
         settings = Settings.create_from_envs()
+    assert settings  # nosec
 
     logging.basicConfig(level=settings.LOG_LEVEL.value)
     logging.root.setLevel(settings.LOG_LEVEL.value)
+    for package in ["aiocache", "botocore", "hpack"]:
+        logging.getLogger(package).setLevel(logging.WARNING)
```
Comment (pcrespov) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py:33]: i see some hacking here to get "noise" down :-) 
The only MINOR issue here is that if one day these libraries are not anymore there the loggers will not exists but you are unnecessarily creating them.

File: services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py [hunk 4]
```diff
@@ -57,4 +61,7 @@ def create_app(settings: Optional[Settings] = None) -> FastAPI:
     if settings.DATCORE_ADAPTER_TRACING:
         setup_tracing(app, settings.DATCORE_ADAPTER_TRACING)
 
+    app.add_exception_handler(HTTPException, http_error_handler)
```
Comment (pcrespov) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py:64]: thx for these updates!

File: services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py [hunk 5]
```diff
@@ -52,9 +35,50 @@ def _compute_file_path(
     return Path(file_path)
 
 
+class PennsieveAuthorizationHeaders(TypedDict):
+    Authorization: str
+
+
+_TTL_CACHE_AUTHORIZATION_HEADERS_SECONDS: Final[
+    int
+] = 3530  # pennsieve authorizes 3600 seconds
```
Comment (pcrespov) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py:44]: MINOR: add here reference to pennsieve relevant doc? you also mention recipes .. add refs here please

**Review by pcrespov — APPROVED** [review]
👍 

You already did some useful upgrades on this service. Can I persuade you to do one extra "maintenance" PR follow this one and
-  Add ``typecheck`` step (see ``unit-test-postgres-database`` for referece)
-  Do full update of reqs. (some constraints were added due to pennsieve lib)

I suggest them in a separate PR for clarity

File: services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py (hunk 3 above)
Comment (sanderegg) [reply services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py:33]: I am using the same logic as you did in the webserver now. but I guess one way to not create loggers would be something like they describe [here](https://www.roelpeters.be/check-if-python-logger-already-exists/) but I'm not so sure whether it is really an issue...

**Comment by sanderegg** [comment]
> +1
> 
> You already did some useful upgrades on this service. Can I persuade you to do one extra "maintenance" PR follow this one and
> 
> * Add `typecheck` step (see `unit-test-postgres-database` for referece)
> * Do full update of reqs. (some constraints were added due to pennsieve lib)
> 
> I suggest them in a separate PR for clarity

@pcrespov :+1: 
1. requirements are already on the latest as far as I can tell
2. added the typechecking...

File: services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py (hunk 3 above)
Comment (pcrespov) [reply services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py]: No, it is I said something minor!

File: services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py [hunk 6]
```diff
@@ -1,38 +1,44 @@
...
 @router.get(
     "/datasets",
     summary="list datasets",
     status_code=status.HTTP_200_OK,
     response_model=Page[DatasetsOut],
 )
-@cancellable_request
+@cancel_on_disconnect
+@cached(
+    ttl=_PENNSIEVE_CACHING_TTL_S,
```
Comment (GitHK) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py:31]: Question, why only 5 minutes? What was the reasoning behind it. Pleas note, I don't know how this data is used usually. I think it's just displayed in the UI, right?

File: services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py [hunk 7]
```diff
@@ -19,42 +19,75 @@
...
     return Path(file_path)
 
 
+class PennsieveAuthorizationHeaders(TypedDict):
+    Authorization: str
+
+
+_TTL_CACHE_AUTHORIZATION_HEADERS_SECONDS: Final[
+    int
+] = 3530  # NOTE: observed while developing this code, pennsieve authorizes 3600 seconds, so we cache a bit less
```
Comment (GitHK) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py:44]: Question: what happens if this expires?

**Review by GitHK — APPROVED** [review]
Nice.
Please find some questions below.

File: services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py (hunk 6 above)
Comment (sanderegg) [reply services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py:31]: Pennsieve is accessible on [https://app.pennsieve.io/](https://app.pennsieve.io/)
- you have access as well
- the data there changes over time and that is outside of our control, hence at the moment I put 5 minutes only
- one possible use-case is that a user uploads some data there, and then wants to use it in osparc, therefore I thought that more than 5 minutes might be a bit annoying

File: services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py (hunk 7 above)
Comment (sanderegg) [reply services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py:44]: good catch. I think I need to rework that thing

Comment (sanderegg) [reply services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py:50]: done

**Pull request merged**